webui/react/src/hooks/useModal/useModalExperimentCreate.tsx

```diff
@@ -330,10 +335,11 @@ const useModalExperimentCreate = (props?: Props): ModalHooks => {
    * title, and buttons, update the modal
    */
   useEffect(() => {
-    if (modalRef.current || (!prevModalProps?.visible && modalProps?.visible)) {
-      openOrUpdate(modalProps);
-    }
-  }, [ modalProps, modalRef, openOrUpdate, prevModalProps ]);
+    if (modalState === prevModalState || !modalState.visible) return;
+
+    const modalProps = getModalProps(modalState);
+    openOrUpdate(modalProps);
```

Comment:
The main thing this change does is, instead of relying on changes in the `modalProps` to trigger the `openOrUpdate`, it purely relies on `modalState`. This was necessary because doing a `isEqual` comparison on `modalProps` causes major issues. `JSON.stringify` on `modalProps.content` causes a bad complaint about circular dependency. This is probably best anyways to avoid having to compare such large objects.